scripts/mirror.test.js

```diff
@@ -7,136 +7,108 @@ import getMatchingBrowserVersion from './mirror.js';
-      assert.equal(match('safari', '7', 'safari_ios'), '8');
-      assert.equal(match('safari', '8', 'safari_ios'), '8');
-      assert.equal(match('safari', '9', 'safari_ios'), '9');
-      assert.equal(match('safari', '9.1', 'safari_ios'), '9.3');
-      assert.equal(match('safari', '10', 'safari_ios'), '10');
-      assert.equal(match('safari', '10.1', 'safari_ios'), '10.3');
-      assert.equal(match('safari', '11', 'safari_ios'), '11');
-      assert.equal(match('safari', '11.1', 'safari_ios'), '11.3');
-      assert.equal(match('safari', '12', 'safari_ios'), '12');
-      assert.equal(match('safari', '12.1', 'safari_ios'), '12.2');
-      assert.equal(match('safari', '13', 'safari_ios'), '13');
-      assert.equal(match('safari', '13.1', 'safari_ios'), '13.4');
-      assert.equal(match('safari', '14', 'safari_ios'), '14');
-      assert.equal(match('safari', '14.1', 'safari_ios'), '14.5');
-      assert.equal(match('safari', '15', 'safari_ios'), '15.1');
-      assert.equal(match('safari', '15.1', 'safari_ios'), '15.1');
-      assert.equal(match('safari', '15.2', 'safari_ios'), '15.2');
-      assert.equal(match('safari', '15.3', 'safari_ios'), '15.3');
-      assert.equal(match('safari', '15.4', 'safari_ios'), '15.4');
-    });
-
-    it('samsunginternet_android', () => {
-      assert.equal(
-        match('chrome_android', '18', 'samsunginternet_android'),
-        '1.0',
-      );
-      assert.equal(
-        match('chrome_android', '25', 'samsunginternet_android'),
-        '1.5',
-      );
-      assert.equal(
-        match('chrome_android', '28', 'samsunginternet_android'),
-        '1.5',
-      );
-      assert.equal(
-        match('chrome_android', '29', 'samsunginternet_android'),
-        '2.0',
-      );
-      assert.equal(
-        match('chrome_android', '33', 'samsunginternet_android'),
-        '2.0',
-      );
-      assert.equal(
-        match('chrome_android', '34', 'samsunginternet_android'),
-        '2.0',
-      );
-      assert.equal(
-        match('chrome_android', '35', 'samsunginternet_android'),
-        '3.0',
-      );
-      assert.equal(
-        match('chrome_android', '95', 'samsunginternet_android'),
-        '17.0',
-      );
-    });
-
-    it('webview_android', () => {
-      assert.equal(match('chrome_android', '18', 'webview_android'), false);
-      assert.equal(match('chrome_android', '25', 'webview_android'), false);
-      assert.equal(match('chrome_android', '26', 'webview_android'), false);
-      assert.equal(match('chrome_android', '27', 'webview_android'), false);
-      assert.equal(match('chrome_android', '28', 'webview_android'), '4.4');
-      assert.equal(match('chrome_android', '36', 'webview_android'), '37');
-      assert.equal(match('chrome_android', '37', 'webview_android'), '37');
-    });
+    for (const [browser, versionMap] of Object.entries(mappings)) {
+      it(browser, () => {
+        for (const pair of versionMap) {
```

Comment:
The downside of this is loop is that it makes it harder to investigate a failure, because the line is the same for all test cases.

I think jest allows parameterised tests or giving each case a description. Not sure if it's worth it?

Reply:
I think a combination of [`describe.each`](https://jestjs.io/docs/api#describeeachtablename-fn-timeout) and [`it.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) would work here.

Reply:
This should do:

```js
describe.each(Object.entries(mappings))('%s', (browser, pairs) => {
  it.each(pairs)('%s, %s', (a, b) => assert.equal(getMatchingBrowserVersion(browser, a, b));
});

Reply:
It looks like Mocha doesn't have `.each()` (we're not using Jest here), but that code looks much nicer and simpler!

Would you mind opening up an issue about switching from Mocha to Jest?  I'm not opposed to switching over to maintain tooling consistency, but that's beyond the scope of this PR.  :P